libpmem2/index.md

```diff
@@ -0,0 +1,209 @@
+---
+title: libpmem2
+layout: pmdk
+---
+
+#### The libpmem2 library
+
+**libpmem2** provides low-level persistent memory support.
+In particular, support for the persistent memory instructions
+for flushing changes to pmem is provided.
+
+This library is provided for software that tracks every store
+to pmem and needs to flush those changes to durability.  Most
+developers will find higher level libraries like
+[libpmemobj](../libpmemobj) to be much more convenient.
+
+Man pages that contains a list of the **Linux** interfaces provided:
+
+* Man page for <a href="../manpages/linux/master/libpmem2/{{ page.title }}.7.html">{{ page.title }} current master</a>
+
+Man pages that contains a list of the **Windows** interfaces provided:
+
+* Man page for <a href="../manpages/windows/master/libpmem2/{{ page.title }}.7.html">{{ page.title }} current master</a>
+
+#### libpmem2 Example
+
+**The Basics**
+
+If you've decided to handle persistent memory allocation
+and consistency across program interruption yourself, you will
+find the functions in libpmem2 useful. It is important to
+understand that programing to raw pmem means you must create
+your own transactions or convince yourself you don't care if
+a system or program crash leaves your pmem files in an inconsistent
+state. Interfaces in the libpmem2 are **non-transactional**, but
+can be used to build transactional interfaces like [libpmemobj](../libpmemobj).
+
+To illustrate the basics, let's walk through the man page example first:
+
+<code data-gist-id='wlemkows/c7dc06875ee4aa9857020eba01114e75' data-gist-file='basic.c' data-gist-line='8-18' data-gist-highlight-line='18' data-gist-hide-footer='true'></code>
+
+The example starts, as shown above, by including the necessary
+headers.  Line 18 (the highlighted line) shows the header file
+you need to include to use libpmem2: `libpmem2.h`.
+
+For this simple example, we're just going to write a simple string to a
+memory-mapped file. First, we need to prepare a config and mapping
+based on the source file.
+
+<code data-gist-id='wlemkows/c7dc06875ee4aa9857020eba01114e75' data-gist-file='basic.c' data-gist-line='20-58' data-gist-highlight-line='39,44,49,55' data-gist-hide-footer='true'></code>
+
+The lines above create the file, prepare config and source structures,
+set required granularity and map the file into memory. 
+
+This illustrates a bunch of the basics functions in libpmem2:
+First of them, `pmem2_config_new()`, create config struct that
+will be used for mapping. In this example, we only set the granularity
+in the config (line 49), other values remain default.
+However, libpmem2 provides multiple functions to configure target mapping,
+eg. `pmem2_config_set_length()` to set length which will be used for mapping,
+or `pmem2_config_set_offset` which will be used to map the contents from the
+specified location of the source.
+
+Second highlighted line 44 above call `pmem2_source_from_fd()`
```

Comment:
I don't think we should use exact line numbers.

This needs a comma:

The second highlighted line contains a call to `pmem2_source_from_fd()`, which takes ...